src/components/Form/FormInput/index.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { forwardRef, KeyboardEvent } from 'react'
+import { useImperativeHandle } from 'react'
+import { ReactNode } from 'react'
+import { useCallback } from 'react'
+import { useRef } from 'react'
+import { ChangeEvent, MutableRefObject } from 'react'
+import Styled, { TextAlign } from './index.style'
+
+interface FormInputProps {
+  id?: string
+  name?: string
+  ref?: MutableRefObject<HTMLInputElement | null>
+  placeholder?: string
+  type?: 'number' | 'password' | 'text'
+  max?: number
+  maxLength?: number
+  textAlign?: TextAlign
+  numberOnly?: boolean
+  value?: string
+  backgroundColor?: string
+  onChange?: (event: ChangeEvent<HTMLInputElement>) => void
+}
```

Comment:
```suggestion
interface FormInputProps extends InputHTMLAttributes<HTMLInputElement>{
  ref?: MutableRefObject<HTMLInputElement | null>
  textAlign?: TextAlign
  numberOnly?: boolean
  backgroundColor?: string
  onChange?: (event: ChangeEvent<HTMLInputElement>) => void
}
```
InputElement 타입 상속을 받아도 좋을 것 같네요